usvm-ts/src/main/kotlin/org/usvm/machine/interpreter/TSInterpreter.kt

```diff
@@ -89,11 +89,16 @@ class TSInterpreter(
     private fun visitIfStmt(scope: TSStepScope, stmt: EtsIfStmt) {
         val exprResolver = exprResolverWithScope(scope)
 
-        val boolExpr = exprResolver.resolve(stmt.condition)?.asExpr(ctx.boolSort)
-            ?: run {
-                logger.warn { "Failed to resolve condition: $stmt" }
-                return
-            }
+        val conditionExpr = exprResolver.resolve(stmt.condition) ?: run {
+            logger.warn { "Failed to resolve condition: $stmt" }
+            return
+        }
+
+        val boolExpr = if (conditionExpr.sort == ctx.boolSort) {
+            conditionExpr.asExpr(ctx.boolSort)
+        } else {
+            ctx.mkTruthyExpr(conditionExpr, scope)
```

Comment:
Is it truly `truthy` or it will be reversed as the condition? For example, not `if (x)`, but `if (!x)`. If we sure that `!` forces value to be boolean, is it a reachable branch?

Reply:
`!` indeed seems to be forcing value to become `boolean`, however not all if-stmts have their conditions reversed. It might be a general observed behavior of the current state of ArkIR and ABC (so that this `mkTruthy` branch is indeed unreachable for "real" code), yet nobody can guarantee it. Moreover, non-negated (plain `if (x)`) conditions arise from the manually constructed IR, like in the current tests (and I do believe it is the correct way here, since it allows for more predictable testing), that's why I even implemented this branch to avoid mismatched sort here.

Overall, it does not hurt anybody to have this "correct" truthy-based impl here, even if it will be "unreachable" on the "real" code.

Reply:
My main concern is about consistency of conditions and branches, so we need to be sure that with `if (x)` condition we'll have the correct `positive` and `negative` branches

Reply:
Both `conditionExpr.asExpr(ctx.boolSort)` and `ctx.mkTruthyExpr(conditionExpr, scope)` represent "positive" branch. And both of them have `boolSort`, so the "negative" branch is going to be a simple negation of it. Isn't it?